src/app/(proper_react)/(redesign)/(authenticated)/user/(dashboard)/dashboard/[[...slug]]/page.tsx

```diff
@@ -103,6 +104,9 @@ export default async function DashboardPage({ params, searchParams }: Props) {
   }
 
   const latestScan = await getLatestOnerepScanResults(profileId);
+  const dataBrokersWithRemovalUnderMaintenance =
+    await getScanResultsWithBrokerUnderMaintenance(profileId);
```

Comment:
@mansaj @codemist I might be a bit late to the discussion, but I’m wondering what your thoughts are on this:

The naming `getScanResultsWithBrokerUnderMaintenance` seems to be a bit misleading. We are only fetching the scan results that are under maintenance here. We have to do this in multiple places all over the application and I’m wondering if we should not have an endpoint that gives us the results with the correct status that we are caring about. Otherwise, we need to fetch and check in place which result has a status mismatch.

Reply:
@flozia are you suggesting we merge this function into the existing GET call for `getLatestOnerepScanResults`?

I don't know if we'd want to do that because we might want to just a "clean" GET for all the scan results for a user

Reply:
Alternatively, we can also have a more barebone function that just fetches the list of all data broker statuses from Redis (fallback to db), but then the frontend would have to do even more comparison / logic

Reply:
Oh, another alternative is to have a function that doesn't do the filtering at all, but just do the JOIN, then just let the frontend have the entire joined scan results (with databroker statuses) and do the filtering whenever/however needed

This might actually be the most flexible (and equally performant) option

@rhelmer @flozia @codemist  wdyt

Reply:
@mansaj I thought that we could maybe not merge it into the existing `getLatestOnerepScanResults`, but rather have one (additional function) that has the joined scan results that we can rely on to have the relevant statuses for the frontend. I like the suggestion of not filtering the results after fetching them as suggested. What do you think @codemist — would this make things easier or more complicated for how the manual broker removal is intended to work?